js/src/builder/struct.ts

```diff
@@ -25,11 +25,18 @@ import { Struct, TypeMap } from '../type.js';
 export class StructBuilder<T extends TypeMap = any, TNull = any> extends Builder<Struct<T>, TNull> {
     public setValue(index: number, value: Struct<T>['TValue']) {
         const children = this.children;
-        switch (Array.isArray(value) || value.constructor) {
-            case true: return this.type.children.forEach((_, i) => children[i].set(index, value[i]));
-            case Map: return this.type.children.forEach((f, i) => children[i].set(index, value.get(f.name)));
-            default: return this.type.children.forEach((f, i) => children[i].set(index, value[f.name]));
+            switch (Array.isArray(value) || value.constructor) {
+                case true: return this.type.children.forEach((_, i) => children[i].set(index, value[i]));
+                case Map: return this.type.children.forEach((f, i) => children[i].set(index, value.get(f.name)));
+                default: return this.type.children.forEach((f, i) => children[i].set(index, value[f.name]));
+            }
```

Comment:
Please un-indent the block so it's aligned with the previous line.

Reply:
That was sloppy, apologies, had whitespace changes off on the diff and didn't notice. 
I've fixed it.